source/globalCommands.py

```diff
@@ -3284,6 +3286,40 @@ def script_braille_cycleShowMessages(self, gesture: inputCore.InputGesture) -> N
 		msg = _("Braille show messages %s") % ShowMessages(newValue).displayString
 		ui.message(msg)
 
+	@script(
+		# Translators: Input help mode message for cycle through braille show selection command.
+		description=_("Cycle through the braille show selection states"),
+		category=SCRCAT_BRAILLE
+	)
+	def script_braille_cycleShowSelection(self, gesture: inputCore.InputGesture) -> None:
+		"""Set next state of braille show selection and reports it with ui.message."""
+		featureFlag: FeatureFlag = config.conf["braille"]["showSelection"]
+		boolFlag: BoolFlag = featureFlag.enumClassType
+		values = [x.value for x in boolFlag]
+		currentValue = featureFlag.value.value
+		nextName: str = boolFlag(
+			currentValue % len(values) + 1
+		).name
+		config.conf["braille"]["showSelection"] = nextName
+		featureFlag = config.conf["braille"]["showSelection"]
+		if featureFlag.isDefault():
+			displayString = featureFlag.behaviorOfDefault.displayString
+			# Translators: Used when reporting braille show selection feature flag
+			# default behavior.
+			msg = _("Braille show selection")
+			msg += f" {self._featureFlagDefaultBehaviorDisplayString(displayString)}"

```

Comment:
This should be a single string so translators can re-arrange the parts.

```suggestion
			msg = _("Braille show selection default (%s)") % featureFlag.behaviorOfDefault.displayString
```

Reply:
Why not a format string?

```suggestion
			msg = _("Braille show selection default ({})").format(featureFlag.behaviorOfDefault.displayString)
```

Reply:
I think that would be also possible. But why not "f style" with one msg assignment (asking this just for curiosity).

Reply:
gettext (the translation library) doesn't support f strings